### DISCUSSION TITLE
Symfony and Hotwire Turbo

### DISCUSSION THREAD

**Discussion by chrishemmings** [coreui/coreui#859] [Q&A]
Hey there,

I've built myself a skeleton Symfony application with CoreUI, which is awesome and looks amazing.  I'm currerently leaning on the Symfony UI stack, so, am using AssetMapper and Stimulus controllers to look after everything.   The next thing on my list is to switch on Hotwire Turbo to make everything feel a whole lot faster.  This is where things have started to fall apart.

So, on my first page load, which uses one of the example templates, everything works as expected.  When I navigate to a new page, which Turbo deals with (i.e loads in the content and replaces <body> with the loaded in content) things like menu's and menu collapse buttons stop working.

Is there a way to re-initilise CoreUI so event listeners can re-attach to the replacement body content?

Thanks :-)

**Comment by artmit** [comment]
I am not sure if you resolved that problem - but I think it can be relate to this: https://symfonycasts.com/screencast/turbo/modal-preview

**Reply by chrishemmings** [reply]
It gives a general idea as to what the problem is, for sure.

I think, what would be great is to have a way to reintialize certain CoreUI functionality.  Like for example, tooltips that stop working because the elements have been replaced.  Or, expanding/dropdown menus that don't work, again because the elements have been replaced.

An example could be something like this..

```javascript
  // Reinitialise tooltips.
  document.querySelectorAll('[data-coreui-toggle="tooltip"]').forEach(el => {
    new coreui.Tooltip(el);
  });

  // Reinitialise popovers.
  document.querySelectorAll('[data-coreui-toggle="popover"]').forEach(el => {
    new coreui.Popover(el);
  });

  // Similarly, add any additional reinitialisation code for dropdowns, collapses, etc.
```
